minicluster/src/main/java/alluxio/underfs/UnderFileSystemCluster.java

```diff
@@ -131,7 +135,20 @@ public UnderFileSystemCluster(String baseDir) {
    * expected to be called either before or after each test case which avoids certain overhead from
    * the bootstrap.
    */
-  public abstract void cleanup() throws IOException;
+  public void cleanup() throws IOException {
+    if (isStarted()) {
+      String path = getUnderFilesystemAddress() + AlluxioURI.SEPARATOR;
+      UnderFileSystem ufs = UnderFileSystem.Factory.get(path);
+      for (UnderFileStatus p : ufs.listStatus(path)) {
+        String childPath = PathUtils.concatPath(path, p.getName());
+        if (p.isDirectory()) {
+          ufs.deleteDirectory(childPath, DeleteOptions.defaults().setRecursive(true));
+        } else {
+          ufs.deleteFile(childPath);
+        }
+      }
+    }
+  }
```

Comment:
I prefer to keep this an abstract method and make it explicit that each under storage should handle clean up differently, since this default cleanup does not seem like it will work well for all types of under storages.